src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2751,6 +2751,34 @@ private bool isMoreSpecificType(CORINFO_CLASS_STRUCT_* cls1, CORINFO_CLASS_STRUC
             return merged == type1;
         }
 
+        private TypeCompareState isEnum(CORINFO_CLASS_STRUCT_* cls, CORINFO_CLASS_STRUCT_** underlyingType)
+        {
+            if (underlyingType != null)
+            {
+                *underlyingType = null;
+            }
+
+            if (cls == null)
```

Comment:
Make sure to do matching changes here once you incorporate feedback for the VM implementation